src/fmu/dataio/dataio.py

```diff
@@ -787,16 +763,24 @@ def generate_metadata(
 
         self._update_check_settings(kwargs)
 
+        if isinstance(obj, (str, Path)):
+            assert self.casepath is not None
+            _future_warning_preprocessed()
+            return deepcopy(
```

Comment:
Here you return a  `deepcopy()` but later (in `export()` not. Any reasons for that?

Reply:
Mainly because the export returns a `str` with the path to the exported file, while the `generate_metadata` returns a dictionary with the metadata.